src/table/table.component.tsx

```diff
@@ -78,13 +128,43 @@ const Table = React.memo((props: TableProps): React.ReactElement => {
   );
 
   const tableInstance = useTable(
-    { columns: displayedColumns, data, defaultColumn },
+    {
+      columns: checkedColumns,
+      data,
+      defaultColumn,
+    },
     useResizeColumns,
-    useFlexLayout
+    useFlexLayout,
+    useColumnOrder
   );
 
-  const { getTableProps, getTableBodyProps, headerGroups, rows, prepareRow } =
-    tableInstance;
+  const {
+    getTableProps,
+    getTableBodyProps,
+    headerGroups,
+    rows,
+    state,
+    prepareRow,
+    setColumnOrder,
+    visibleColumns,
+  } = tableInstance;
+
+  const { columnOrder } = state;
+
+  const updater = (result: any): string[] => {
```

Comment:
The name of this function is very generic to the point it's hard to tell what it's for. Can we rename this something more verbose? colOrderUpdater or something??